federator/src/lib/TransactionSender.js

```diff
@@ -37,11 +37,18 @@ module.exports = class TransactionSender {
         return this.getEthGasPrice();
     }
 
-    async getGasLimit() {
+    async getGasLimit(rawTx) {
+        const estimatedGas = await this.client.eth.estimateGas({
+            gasPrice: rawTx.gasPrice,
+            value: rawTx.value,
+            to: rawTx.to,
+            data: rawTx.data,
+            from: rawTx.from
+        });
         // Gas estimation does not work correctly on RSK and after the London harfork, neither is working on Ethereum
```

Comment:
we should remove these comments right?

Reply:
Oh no, we need that, other wise we wouldn't know why we are harcoding that and we will make the same mistake again

Reply:
yeah what I mean is that we're not hardcoding it anymore - we only default to `ESTIMATED_GAS` if the estimated gas is below a certain threshold, right?
maybe just ignore this comment haha